ci/workflow_summary/workflow_information.py

```diff
@@ -0,0 +1,250 @@
+
+
+'''A utility collecting ci_test.yml workflow failure logs.
+
+Usage:
+
+  python workflow_information.py --token ${github_toke} --workflow_name ${workflow_name}
+
+'''
+
+def main():
+  logging.getLogger().setLevel(logging.INFO)
+
+  args = parse_cmdline_args()
+  logging.info(args)
+
+  gh = github.GitHub(args.repo_owner, args.repo_name)
+
+  # location for all artifacts
+  if args.folder:
+    file_folder = os.path.normpath(args.folder)
+  else:
+    file_folder = os.path.normpath(datetime.datetime.utcnow().strftime('%Y-%m-%d+%H:%M:%S'))
+  if not os.path.exists(file_folder):
+    os.makedirs(file_folder)
+
+  workflow_summary = get_workflow_summary(gh, args)
+  workflow_summary_file_path = os.path.join(file_folder, 'workflow_summary.json')
+  with open(workflow_summary_file_path, 'w') as f:
+    json.dump(workflow_summary, f)
+  logging.info(f'Workflow summary has been write to {workflow_summary_file_path}\n')
+
+  job_summary = get_job_summary(workflow_summary)
+  job_summary_file_path = os.path.join(file_folder, 'job_summary.json')
+  with open(job_summary_file_path, 'w') as f:
+    json.dump(job_summary, f)
+  logging.info(f'Job summary has been write to {job_summary_file_path}\n')
+
+  workflow_summary_report = f"{datetime.datetime.utcnow()}\n{args}\n\n"
+  workflow_summary_report += generate_summary_report(workflow_summary, job_summary)
+  report_file_path = os.path.join(file_folder, 'workflow_summary_report.txt')
+  with open(report_file_path, 'w') as f:
+    f.write(workflow_summary_report)
+  logging.info(f'Workflow summary report has been write to {report_file_path}\n')
+
+
+def get_workflow_summary(gh, args):  
+  token = args.token
+  workflow_name = args.workflow_name
+  # https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax#query-for-dates
+  days = args.days
+  current_datetime = datetime.datetime.utcnow()
+  since_datetime = current_datetime - datetime.timedelta(days=days)
+  created = '>' + since_datetime.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+  workflow_summary = {'workflow_name': workflow_name, 
+                    'total_count': 0, 
+                    'success_count': 0, 
+                    'failure_count': 0, 
+                    'created': created,
+                    'workflow_runs': []}
+
+  logging.info('START collecting workflow run data\n')
+  workflow_page = 0
+  per_page = 100 # max 100
+  list_workflows_params = {'status': 'completed', 'created': created, 'page': workflow_page, 'per_page': per_page}
+  if args.event:
+    list_workflows_params['event'] = args.event
+  if args.actor:
+    list_workflows_params['actor'] = args.actor
+  if args.branch:
+    list_workflows_params['branch'] = args.branch
+
+  while True:
+    workflow_page += 1
+    list_workflows_params['page'] = workflow_page
+    workflows = gh.list_workflows(token, workflow_name, list_workflows_params)
+
+    if 'workflow_runs' in workflows and workflows['workflow_runs']:
```

Comment:
It will improve readability if you invert the conditional:

```
while True:
  ...
  if 'workflow_runs' not in workflows or not workflows['workflow_runs']:
    break
  ...
```
This will reduce nesting